adapters/socketio_berkeley.c

```diff
@@ -946,7 +946,7 @@ int socketio_send(CONCRETE_IO_HANDLE socket_io, const void* buffer, size_t size,
                         /* queue data */
                         size_t bytes_sent = (send_result < 0 ? 0 : send_result);
 
-                        if (add_pending_io(socket_io_instance, buffer + bytes_sent, size - bytes_sent, on_send_complete, callback_context) != 0)
+                        if (add_pending_io(socket_io_instance, (unsigned char*)buffer + bytes_sent, size - bytes_sent, on_send_complete, callback_context) != 0)
```

Comment:
nit: we could add `const` as all definitions have it (https://sourcegraph.com/github.com/Azure/azure-c-shared-utility@003b9d79ffc91debd85a9ccc12a24c80ccf884e4/-/blob/adapters/socketio_berkeley.c#L949:29&tab=def)